app/forms/work_packages/dialogs/create_form.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+
+module WorkPackages::Dialogs
+  class CreateForm < ApplicationForm
+    include CustomFields::CustomFieldRendering
+
+    attr_reader :work_package, :wrapper_id, :contract
+
+    def initialize(work_package:, wrapper_id:)
+      super()
+
+      @work_package = work_package
+      @wrapper_id = wrapper_id
+      @contract = WorkPackages::CreateContract.new(work_package, User.current)
+    end
+
+    form do |f|
+      f.autocompleter(
+        name: :type_id,
+        required: true,
+        include_blank: false,
+        input_width: :small,
+        label: Type.model_name.human,
+        visually_hide_label: true,
+        autocomplete_options: {
+          multiple: false,
+          decorated: true,
+          clearable: false,
+          focusDirectly: false,
+          hiddenFieldAction: "change->work-packages--create-dialog#refreshForm",
+          append_to: wrapper_id,
+          data: { test_selector: "work_package_create_dialog_type" }
+        }
+      ) do |select|
+        contract
+          .assignable_types
+          .pluck(:id, :name)
+          .map do |value, label|
+          select.option(label:,
+                        value:,
+                        classes: "__hl_inline_type_#{value}",
+                        selected: work_package.type_id == value)
+        end
+      end
+
+      f.text_field(
+        name: :subject,
+        label: WorkPackage.human_attribute_name(:subject),
+        required: true,
+        autofocus: true,
```

Comment:
Field should be focused only if there are no errors in the model. If there are errors, then the first errored field should be focused (which should be done automatically by Primer forms).

Reply:
And I tested it and it does focus on the field in error, except for rich text field. So it makes it hard to see why clicking "Create" button does not work as the rich text input field having the error is not scrolled to.

Reply:
To focus the CKEditor field, we would need to pass the input some flag that it's invalid AND that it's the first field. This is going to be hard to do, and having a required long text field is probably an edge case. I'd ignore that, accepting that people have to click into the field.

Reply:
yep, seems a good trade-off.